### ISSUE TITLE
Build fail : Segmentation fault (core dumped) /usr/bin/ld [...]

### ISSUE THREAD

**Issue by ghost** [monero-project/monero#1131]
Build env : raspberry pi / arch linux

make[3]: Entering directory '/home/alarm/tmp/monero-0.10.0/build/release'
[ 50%] Generating testnet_blocks.o
/bin/sh: line 1: 8895 Segmentation fault (core dumped) /usr/bin/ld -r -b binary -o /home/alarm/tmp/monero-0.10.0/build/release/src/blocks/testnet_blocks.o testnet_blocks.dat
make[3]: **\* [src/blocks/CMakeFiles/blocks.dir/build.make:66: src/blocks/testnet_blocks.o] Error 139

... too bad, I realy want to have a full node running.

Rgds

**Comment by anonimal** [comment]
@lotusfred have you tried cloning/building against current master?

Referencing https://aur.archlinux.org/packages/monero/

**Comment by NanoAkron** [comment]
Which RPi?

**Comment by lotusfred** [comment]
@anonimal : Yes I first try to use sources from AUR (and got the seg fault) then I try with "git clone https://github.com/monero-project/monero.git" and got the same error 
By the way I first post my question on https://aur.archlinux.org/packages/monero/ (nikname fredchamp)
@NanoAkron : I try to build on a RPI-2B

**Comment by NanoAkron** [comment]
Please try #1269

**Comment by radfish** [comment]
See #974. Try Invoking cmake with: `-DCMAKE_LINKER=/usr/bin/ld.gold`

**Comment by diederikdehaas** [comment]
I had a problem that very much seems to be related to `ld` vs `ld.gold` too on my Debian Sid amd64 system (haven't tried yet on my RPi's).

I tried to build monero-core and one of the first steps is run `./get_libwallet_api.sh`, which builds part of monero and that failed on the following line:
`ld -r -b binary -o /home/diederik/dev/crypto/monero-project/monero/build/release/src/blocks/testnet_blocks.o testnet_blocks.dat`

But it did succeed when I manually executed:
`ld.gold -r -b binary -o /home/diederik/dev/crypto/monero-project/monero/build/release/src/blocks/testnet_blocks.o testnet_blocks.dat`

A couple of days later the `binutils` package, which provides both `ld` and `ld.gold`, got updated from version `2.27.51.20161201-1` to `2.27.51.20161220-1` and then it did succeed with `ld`.
However, I didn't manage to reproduce the problem on my Debian Strech amd64 system and I also hadn't saved the build log :-/

Another thing to keep in mind is the version of openssl-dev used, 1.0 or 1.1, see https://github.com/monero-project/monero/issues/1411 for details. 
Also note that openssl 1.1 isn't yet supported by the Qt project.

So it may be useful to investigate which versions of various libraries are in use.
